enumerable.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    new_arr = []
+    (0..length - 1).each do |i|
+      new_arr.push(self[i]) if yield(self[i])
+    end
+    new_arr
+  end
+
+  def my_all?(*arg)
+    if block_given?
+      my_each do |i|
+        return false unless yield(i)
+      end
+    elsif !arg.empty?
+      if arg[0].is_a? Numeric
+        my_each do |i|
+          return false if i != arg[0]
+        end
+      elsif arg[0].class == Regexp
+        my_each do |i|
+          return false if i !~ arg[0]
+        end
+      else
+        my_each do |i|
+          return false unless i.is_a? arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        return false if i.nil? || i == false
+      end
+    end
+    true
+  end
+
+  def my_any?(*arg)
+    if block_given?
+      my_each do |i|
+        return true if yield(i)
+      end
+    elsif !arg.empty?
+      if arg[0].is_a? Numeric
+        my_each do |i|
+          return true if i == arg[0]
+        end
+      elsif arg[0].class == Regexp
+        my_each do |i|
+          return true if i =~ arg[0]
+        end
+      elsif arg[0].is_a? String
+        my_each do |i|
+          return true if i == arg[0]
+        end
+      else
+        my_each do |i|
+          return true if i.is_a? arg[0]
+        end
+      end
+    else
+      return false if empty?
+
+      my_each do |i|
+        return true if !i.nil? && i != false
+      end
+    end
+    false
+  end
+
+  def my_none?(*arg)
+    if block_given?
+      puts "block given"
+      my_each do |i|
+        return false if yield(i)
+      end
+    elsif !arg.empty?
+      puts "arg given"
+      if arg[0].is_a? Numeric
+        my_each do |i|
+          return false if i == arg[0]
+        end
+      elsif arg[0].class == Regexp
+        puts "Regexp"
+        my_each do |i|
+          return false if arg[0] =~ i
+        end
+      else
+        my_each do |i|
+          return false if i.is_a? arg[0]
+        end
+      end
+    else
+      return true if empty?
+
+      my_each do |i|
+        return false if i
+      end
+    end
+    true
+  end
+
+  def my_count(num = nil)
+    count = 0
+    if block_given?
+      (0..length - 1).each do |i|
+        next unless yield(self[i])
+
+        count += 1
+      end
+    elsif num
+      my_each do |i|
+        count += 1 if i == num
+      end
+    else
+      my_each do |_i|
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_inject(*args)
+    sum = 0
```

Comment:
my_inject is not working right now:
`block in my_inject': undefined method 'zero' for 0:Integer (NoMethodError)`

Remember, my_inject should work as the original method:
Check inject [documentation](https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-inject).

Reply:
Yes, sorry that is a fix from rubocop who takes down ==0 to pleace with that instead, but doesnt work, this rubocop is driving me crazy. But found out that is a bug of the lintern where is highly recommend turn off the lintern.